pkg/backend/openshiftcluster.go

```diff
@@ -197,19 +198,41 @@ func (ocb *openShiftClusterBackend) updateAsyncOperation(ctx context.Context, lo
 			now := time.Now()
 			asyncdoc.AsyncOperation.EndTime = &now
 
+			// Returns a generic error
+			handleOtherError := func() {
+				log.Error(backendErr)
+				asyncdoc.AsyncOperation.Error = &api.CloudErrorBody{
+					Code:    api.CloudErrorCodeInternalServerError,
+					Message: "Internal server error.",
+				}
+			}
+
 			if provisioningState == api.ProvisioningStateFailed {
-				// if type is CloudError - we want to propagate it to the
-				// asyncOperations errors. Otherwise - return generic error
-				err, ok := backendErr.(*api.CloudError)
-				if ok {
+				switch err := backendErr.(type) {
+				case *api.CloudError:
+					// if type is CloudError - we want to propagate it to the asyncOperations errors.
 					log.Print(backendErr)
 					asyncdoc.AsyncOperation.Error = err.CloudErrorBody
-				} else {
-					log.Error(backendErr)
+				case *azure.RequestError:
```

Comment:
What is the API call that gets this error and can we handle the error more closely to it rather than here?  For example, we already handle miscellaneous ServiceErrors in deployARMTemplate, returning CloudErrorCodeDeploymentFailed and the error text.  Can we use that kind of handler on the API call in question?

Reply:
It is a call to CreateOrUpdate on the cluster manager's ResourceGroupsClient. I've moved it there and changed the code to CloudErrorCodeDeploymentFailed.